### PR TITLE
Update db

### PR DESCRIPTION
**Pull request by levy002**
## Description
In this PR, we updated The Vet_clinic database.

- [x] Followed the **Github flow**.
- [x] The documentation is professional.

### Milestone requirements, We:

- [x]  added a column `species` of type `string` to `animals` table. 
- [x] inserted new data.
- [x]  updated the table by `BEGIN` a transaction and update the `animals` table by setting the `species` column to `unspecified` and `ROLLBACK`. 

- [x]  updated the table by `BEGIN` a transaction and:

> update the `animals` table by setting the `species` column to `digimon` for all animals that have a name ending in `mon`.

> update the `animals` table by setting the `species` column to `pokemon` for all animals that don't have `species` already set.

> `COMMIT` the transaction.

- [x]  updated the table by `BEGIN` a transaction and:

>  `DELETE` all records in the `animals` table.

> `ROLLBACK` the transaction.

- [x]  updated the table by `BEGIN` a transaction and:

> `DELETE` all animals born after Jan 1st, 2022.

> created a `SAVEPOINT` for the transaction.

> updated all animals' weight to be their weight multiplied by -1.

> `ROLLBACK TO SAVEPOINT`.

> Updated all animals' weights that are negative to be their weight multiplied by -1.

> `COMMIT` the transaction.

- [x] wrote queries to answer the following questions: 

> How many animals are there?

> How many animals have never tried to escape?

> What is the average weight of animals?

> Who escapes the most, neutered or not neutered animals?

> What is the minimum and maximum weight of each type of animal?

> What is the average number of escape attempts per animal type of those born between 1990 and 2000?

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -1,4 +1,10 @@
...
+INSERT INTO animals(id, name, date_of_birth, escape_attempts, neutered, weight_kg) values(DEFAULT, 'Gabumon', '2018-11-15', 2, true, 8),(DEFAULT, 'Pikachu', '2021-01-07', 1, false, 15.04),(DEFAULT, 'Devimon', '2017-05-12', 5, true,11);
+INSERT INTO animals(id, name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES (DEFAULT ,'Charmander', '2020-02-08', 0, false, -11);
+INSERT INTO animals(id, name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES (DEFAULT ,'Squirtle', '1993-04-02', 3, false, -12.13);
+INSERT INTO animals(id, name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES (DEFAULT ,'Angemon', '2005-06-12', 3, true, -45);
+INSERT INTO animals(id, name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES (DEFAULT ,'Boarmon', '2005-06-07', 7, true, 20.4);
+INSERT INTO animals(id, name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES (DEFAULT ,'Blossom', '1998-10-13', 3, true, 17);
+INSERT INTO animals(id, name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES (DEFAULT ,'Ditto', '2022-05-14', 4, true, 22);
```
Comment (Kingstalux) [inline data.sql]: - [x] Good job so far 👍 Kindly add the data for `Plantmon` as it is not present in your data.sql

File: queries.sql [hunk 2]
```diff
@@ -16,3 +16,53 @@ select * from animals where neutered =true;
...
+ -- Update all animals' weights that are negative to be their weight multiplied by -1.
+ UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 1;
+ -- Commit transaction
+ COMMIT;
+
+
+-- How many animals are there?
+SELECT count(*) FROM animals;
+-- How many animals have never tried to escape?
+SELECT count(*) FROM animals WHERE escape_attempts > 0;
```
Comment (Kingstalux) [inline queries.sql]: - [x] For animals that have never tried to escape, their `escape_attempts` should be equal to 0 and not greater than 0. Kindly correct that 👍

File: queries.sql [hunk 3]
```diff
@@ -16,3 +16,53 @@ select * from animals where neutered =true;
...
+-- What is the average weight of animals?
+SELECT AVG(weight_kg) FROM animals;
+-- Who escapes the most, neutered or not neutered animals?
+SELECT name FROM animals WHERE escape_attempts = (SELECT MAX(escape_attempts) FROM animals);
```
Comment (Kingstalux) [inline queries.sql]: - [x] This query is to get the group of animals that escapes the most and not to get the name of the animal with the highest escape attempt. Kindly make use of `GROUP BY` to correct this query 💯

File: queries.sql [hunk 4]
```diff
@@ -16,3 +16,53 @@ select * from animals where neutered =true;
...
+-- What is the minimum and maximum weight of each type of animal?
+SELECT MAX(weight_kg) FROM animals WHERE species = 'digimon';
+SELECT MAX(weight_kg) FROM animals WHERE species = 'pokemon';
+SELECT MIN(weight_kg) FROM animals WHERE species = 'pokemon';
+SELECT MIN(weight_kg) FROM animals WHERE species = 'digimon';
```
Comment (Kingstalux) [inline queries.sql]: - [x] Kindly make use of `GROUP BY` to shorten these queries. You can group them by species 😄

File: queries.sql [hunk 5]
```diff
@@ -16,3 +16,53 @@ select * from animals where neutered =true;
...
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT AVG(escape_attempts) FROM animals WHERE date_of_birth BETWEEN '1990-12-31' AND '2000-12-31' AND species = 'digimon';
+SELECT AVG(escape_attempts) FROM animals WHERE date_of_birth BETWEEN '1990-12-31' AND '2000-12-31' AND species = 'pokemon';
```
Comment (Kingstalux) [inline queries.sql]: - [x] Same issue here as above. Make use of `GROUP BY` to shorten this query 👍 Take special note that you are to write one query to answer one question and not multiple queries to answer a single question. That is where `GROUP BY` comes in to help. Kindly rectify this 💯

**Review by Kingstalux — CHANGES_REQUESTED** [review]
Hi @levy002, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
68747470733a2f2f6d65646961302e67697068792e636f6d2f6d656469612f7741366159414f336a4836526d55494259732f67697068792e676966


## Highlights 👏 
✅ Good PR description 👍 
✅ Professional README file 💯 
✅ Correct Gitflow used 👌 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Thank you @Kingstalux for the excellent suggestions. I've implemented them. :handshake:

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @levy002,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**